geniza/corpus/management/commands/export_metadata.py

```diff
@@ -101,35 +115,49 @@ def paths(self):
     ## LOCAL EXPORTING
     ############################################
 
-    def export_data(self):
+    def export_data(self, lastrun, sync=False):
         "generate all exports"
-        with self.timer("Exporting metadata into local path"):
-            # make sure to pull first
-            self.repo_pull()
-
-            # write docs
-            with self.timer("Exporting document data"):
-                PublicDocumentExporter(progress=self.progress).write_export_data_csv(
-                    self.path_documents_csv
-                )
-
-            # write fragments
-            with self.timer("Exporting fragment data"):
-                PublicFragmentExporter(progress=self.progress).write_export_data_csv(
-                    self.path_fragments_csv
-                )
-
-            # write sources
-            with self.timer("Exporting source data"):
-                SourceExporter(progress=self.progress).write_export_data_csv(
-                    self.path_sources_csv
-                )
-
-            # write footnotes
-            with self.timer("Exporting footnotes"):
-                FootnoteExporter(progress=self.progress).write_export_data_csv(
-                    self.path_footnotes_csv
-                )
+
+        # pull any remote changes before making any local modifications
+        self.repo_pull()
+
+        # get all log entries since the last run
+        log_entries = LogEntry.objects.filter(action_time__gte=lastrun)
+
+        # if there are NO changes, bail out
+        if log_entries.count() == 0:
+            self.print("No changes since last run")
+            return
+
+        # run all configured exports
+        for export_name, exporter in self.exports.items():
+            with self.timer(f"Exporting { export_name } data"):
+                subset_logentries = log_entries.filter(**exporter.content_type_filter)
+                # if there are no changes, move to next export
+                if not subset_logentries.count():
+                    self.print("No changes for %s" % export_name)
+                    continue
```

Comment:
Ooh v nice! So we only export precisely the dataset that changed.

Reply:
Yes, that's the idea — although it's at a fairly rough level of granularity, and it could be that parts of the model were changed that don't actually impact the export.